gprofiler/metadata/versions.py

```diff
@@ -26,7 +26,14 @@ def get_exe_version(
     def _run_get_version() -> "CompletedProcess[bytes]":
         return run_process([exe_path, version_arg], stop_event=stop_event, timeout=get_version_timeout)
 
-    cp = run_in_ns(["pid", "mnt"], _run_get_version, process.pid)
+    try:
+        cp = run_in_ns(["pid", "mnt"], _run_get_version, process.pid)
+    except FileNotFoundError as e:
+        if not pid_exists(process.pid):
```

Comment:
Pid recycling can take place before this check.
If you use the is_alive (or whatever its called) method on thr process object you already have, psutil will take care of mitigating pid recycle by also checking if the process start time is the the same